### ISSUE TITLE
Fine-grained PAT Support

### ISSUE THREAD

**Issue by chtzvt** [Legit-Labs/legitify#194]
### TL;DR

GitHub [recently announced](https://github.blog/changelog/2023-04-27-graphql-improvements-for-fine-grained-pats-and-github-apps/) that fine-grained PATs can now be used to call the [GitHub GraphQL API](https://docs.github.com/en/graphql).

As a result, it should be possible to use these with Legitify.

### Detailed design

It may be a simple as removing the check in [`validateToken`](https://github.com/Legit-Labs/legitify/blob/468aeb2e73145a10bc6bfb9f69ecdc667d47134b/internal/clients/github/client.go#L113) to enable support for fine-grained PATs. I'll test this and get a better idea if that's the case.

**Comment by chtzvt** [comment]
The following modifications to `internal/clients/github/client.go` cause Legitify to accept fine-grained PATs: 

```go
var githubTokenPattern = regexp.MustCompile("^ghp_[a-zA-Z0-9]{36}$|^github_pat_[a-zA-Z0-9]{22}_[a-zA-Z0-9]{59}$")

func (c *Client) validateToken(token string) error {
	if token == "" {
		return fmt.Errorf("missing token")
	} else if !githubTokenPattern.MatchString(token) {
		return fmt.Errorf("GitHub token seems invalid (expected pattern: '%v')", githubTokenPattern)
	}

	return nil
}
```

Analysis runs fine until we hit a panic on `github.com/Legit-Labs/legitify/cmd/progressbar.(*progressBar).handleTimedBarCreation` with the message, `panic: *mpb.Progress instance can't be reused after it's done!`.

```
 ___      _______  _______  ___   _______  ___   _______  __   __
|   |    |       ||       ||   | |       ||   | |       ||  | |  |
|   |    |    ___||    ___||   | |_     _||   | |    ___||  |_|  |
|   |    |   |___ |   | __ |   |   |   |  |   | |   |___ |       |
|   |___ |    ___||   ||  ||   |   |   |  |   | |    ___||_     _|
|       ||   |___ |   |_| ||   |   |   |  |   | |   |      |   |
|_______||_______||_______||___|   |___|  |___| |___|      |___|
By Legit Security

Using Github Cloud
Note: to get the OpenSSF scorecard results for the organization repositories use the --scorecard option

metadata         6 / 6 [==============================================================] 100 %
organization     1 / 1 [==============================================================] 100 %
actions          0 / 1 [--------------------------------------------------------------] 0 %
repository   121 / 121 [==============================================================] 100 %
panic: *mpb.Progress instance can't be reused after it's done!

goroutine 30 [running]:
github.com/vbauerster/mpb/v8.(*Progress).AddFiller(0x140000ac730, 0x3b, {0x1055bcf60?, 0x1400039eea0?}, {0x140002328e8, 0x1, 0x1})
        /Users/charlton/.go/pkg/mod/github.com/vbauerster/mpb/v8@v8.1.4/progress.go:154 +0x1dc
github.com/vbauerster/mpb/v8.(*Progress).New(0x8?, 0x1400007a900?, {0x12def8008?, 0x14000199000?}, {0x140002328e8, 0x1, 0x1})
        /Users/charlton/.go/pkg/mod/github.com/vbauerster/mpb/v8@v8.1.4/progress.go:125 +0x60
github.com/vbauerster/mpb/v8.(*Progress).AddBar(0x14000bd04e0?, 0x104fa3661?, {0x140002328e8, 0x1, 0x1})
        /Users/charlton/.go/pkg/mod/github.com/vbauerster/mpb/v8@v8.1.4/progress.go:115 +0x118
github.com/Legit-Labs/legitify/cmd/progressbar.(*progressBar).handleTimedBarCreation(0x140000bb7d0, {{0x104fb5d5a, 0x14}, {0xc10ccd44d4e6bd10, 0x1f0ecb3cae, 0x105f85a40}})
        /Users/charlton/Developer/legitify/cmd/progressbar/progressbar.go:190 +0x36c
github.com/Legit-Labs/legitify/cmd/progressbar.(*progressBar).Run.func1()
        /Users/charlton/Developer/legitify/cmd/progressbar/progressbar.go:86 +0x1d8
created by github.com/Legit-Labs/legitify/cmd/progressbar.(*progressBar).Run
        /Users/charlton/Developer/legitify/cmd/progressbar/progressbar.go:72 +0x94
exit status 2
```

**Comment by chtzvt** [comment]
Note that the above results occur for an organization-scoped PAT that is still pending approval. Even so, it shouldn't cause a panic.

**Comment by noamd-legit** [comment]
Hi @chtzvt, Thanks for doing this research!

Since fine-grained tokens don't support the headers you mentioned and non-repository API's It's not trivial to support them now.
**A solution you can try is to support them only for the repository namespace and turn off scope checks (skippers.go).**

We'll support fine-grained tokens when they are more mature.

I am closing the issue for now
